package-res/resources/web/pentaho/lang/SortedList.js

```diff
@@ -0,0 +1,228 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./List",
+  "./Base"
+], function(List, Base) {
+
+  var baseProto = Base.Array.prototype;
+
+  return List.extend("pentaho.lang.SortedList", /** @lends  pentaho.lang.SortedList# */{
+    /**
+     * @classdesc The `SortedList` class is an abstract base class for ordered arrays.
+     *
+     * ### AMD
+     *
+     * To obtain the constructor of this class,
+     * require the module `"pentaho/lang/SortedList"`.
+     *
+     * ### Remarks
+     *
+     * If an attempt is made to add a element to a specific index of the list
+     * an error is thrown.
+     *
+     * @class
+     * @name SortedList
+     * @memberOf pentaho.lang
+     * @abstract
+     * @extends pentaho.lang.List
+     *
+     * @description Initializes a sorted list instance.
+     *
+     * Note that because a `SortedList` is a sub-class of `Array`,
+     * it cannot be a "newable" constructor function (at least up to ECMAScript version 5).
+     * Instead, instances of `SortedList` are actually initial instances of `Array`
+     * whose prototype is then changed to be that of `SortedList`.
+     *
+     * In other words, `SortedList` is an "initialization" constructor
+     * (see {@link pentaho.lang.ISpecifiable} for more information
+     *  on these concepts).
+     *
+     * Concrete `SortedList` sub-classes should provide a static `to` method
+     * to help in their construction.
+     *
+     * @param {Object}   [keyArgs]         The keyword arguments.
+     * @param {Function} [compareFunction] Specifies a function that defines the sort order.
+     *
+     * These are not used directly by the `SortedList` class
+     * but are passed-through to the methods that handle
+     * the initialization of each list element.
+     */
+    constructor: function(keyArgs, compareFunction) {
```

Comment:
Because it doesn't make sense for the elements type.
> These are not used directly by the `SortedList` class but are passed-through to the methods that handle the initialization of each list element.

(also noticed the above sentence belongs near the `keyArgs`... as is it seems to apply to both parameters)